src/Components/PaperEditor/TranscriptsContainer/TranscriptTabContent/index.js

```diff
@@ -116,49 +110,215 @@ const TranscriptTabContent = (props) => {
+      selectedLabels.length > 0
+    ) {
+      setHasSearch(true);
+    } else {
+      setHasSearch(false);
+    }
+
+    return () => {
+      setHasSearch(false);
+    };
+  }, [ searchString, selectedLabels, selectedSpeakers ]);
+
+  useEffect(() => {
+    /* Paragraph text for data attribute for searches, without punctuation */
+    const findString = (str, text) => (str ? text.includes(str) : false);
+
+    const findSpeaker = (speaker) =>
+      !!selectedSpeakers.find((spk) => spk.label === speaker);
+
+    const findLabel = (label) => !!selectedLabels.find((lb) => lb.id === label);
+
+    const isSearchResult = (text, speaker, label) => {
+      const textWithoutPunctuation = removePunctuation(text);
+      const lcSearchString = searchString.toLowerCase();
+
+      const foundSearchString = findString(
+        lcSearchString,
+        textWithoutPunctuation
+      );
+
+      const foundSpeaker = findSpeaker(speaker, selectedSpeakers);
+      const foundLabel = findLabel(label, selectedLabels);
+
+      if (foundSearchString || foundSpeaker || foundLabel) {
+        return true;
+      }
+
+      return false;
+    };
+
+    const isParagraphSearchResult = (paragraph) => {
+      let wordsAnnotations = { labelId: '' };
+      if (annotations) {
+        wordsAnnotations = findAnnotationsInWords(annotations, paragraph.words);
+      }
+
+      if (
+        isSearchResult(
+          paragraph.text,
+          paragraph.speaker,
+          wordsAnnotations.labelId
+        )
+      ) {
+        return true;
+      }
+
+      return false;
+    };
+
+    const setAllDisplays = (display) => {
+      const displayAllParagraphs = paragraphs.map((p) => display);
+      setDisplayParagraphs(displayAllParagraphs);
+    };
+    const setAllSearchResults = (searchResult) => {
+      const searchResults = paragraphs.map((p) => searchResult);
+      setIsSearchResults(searchResults);
+    };
+
+    if (paragraphs) {
+      if (paragraphOnly) {
+        // don't style borders
+        setAllSearchResults(false);
+        if (hasSearch) {
+          const foundParagraphs = paragraphs.map((p) =>
+            isParagraphSearchResult(p)
+          );
+          setDisplayParagraphs(foundParagraphs);
+        } else {
+          setAllDisplays(true);
+        }
+      } else {
+        // display all with styled borders for found paragraphs
+        setAllDisplays(true);
+        if (hasSearch) {
+          const foundParagraphs = paragraphs.map((p) =>
+            isParagraphSearchResult(p)
+          );
+          setIsSearchResults(foundParagraphs);
+        } else {
+          setAllSearchResults(false);
+        }
+      }
+    } else {
+      setAllDisplays(true);
+    }
+
+    return () => {};
+  }, [
+    annotations,
+    hasSearch,
+    paragraphOnly,
+    paragraphs,
+    searchString,
+    selectedLabels,
+    selectedSpeakers,
+  ]);
+
+  const handleSearch = (e) => {
+    if (e.target.type !== 'checkbox') {
```

Comment:
Added this because if `checkbox` is triggered, then the `e.target.value` becomes `on`